libs/agno/agno/models/base.py

```diff
@@ -874,6 +874,33 @@ def _create_function_call_result(
             metrics=MessageMetrics(time=timer.elapsed),
         )
 
+        # Return the message
+        return message
+
+    # This gives the error- agno.exceptions.ModelProviderError: Invalid 'messages[4]'. Image URLs are only allowed for messages with role 'user', but this message with role 'tool' contains an image URL.
```

Comment:
We'll have to create a separate message for "user" with the image, as if the user sent an image after the tool result.  Still not sure whether it would work, but worth a try.